lib/declaration.ts

```diff
@@ -979,32 +1002,44 @@ export default class DeclarationTranspiler extends base.TranspilerBase {
     this.emit('extension');
     visitNameOfExtensions();
     this.emit('on');
-    visitName();
+    visitClassName();
     this.emit('{');
     for (const merged of mergedMembers) {
       const declaration = merged.mergedDeclaration;
-      if (!base.isPromise(declaration.type)) {
-        continue;
-      }
-      this.emit('Future');
-      if (ts.isTypeReferenceNode(declaration.type)) {
-        this.maybeVisitTypeArguments(declaration.type);
+      if (ts.isPropertyDeclaration(declaration) || ts.isPropertySignature(declaration)) {
+        visitTypeAndArguments(declaration.type);
+        this.emit('get');
+        this.visitName(declaration.name);
+        this.emit('{');
+        castThisToPrivateClass();
+        this.emitExtensionGetterBody(declaration);
+        this.emit('}');
```

Comment:
Should I move `this.emit('{')`, `this.emit('}')` and `castToPrivateClass` inside emitExtensionBody?